### PR TITLE
I/5829: Enhance restricted editing feature styling

### PR DESCRIPTION
**Pull request by panr**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Add hover state for all exeptions and fix width for collapsed exception in restricted editing. Closes https://github.com/ckeditor/ckeditor5/issues/5829.

### CODE REVIEW

File: theme/ckeditor5-restricted-editing/restrictedediting.css [hunk 1]
```diff
@@ -36,6 +41,9 @@
 	}
 
 	&.restricted-editing-exception_collapsed {
-		padding-left: 1em;
+		/* Single char is about 10px wide.
+		Value of 8px for collapsed element is because we have additional 2px in left/right borders.
+		Also less than 10px wide is hard to click on target. */
+		padding-left: 8px;
```
Comment (Reinmar) [inline theme/ckeditor5-restricted-editing/restrictedediting.css]: What about `1ch`?

Comment (panr) [reply theme/ckeditor5-restricted-editing/restrictedediting.css]: Good point, totally forgot about that unit 🙉

**Comment by panr** [comment]
> You styled the :hover for both modes – the standard one and the restricted one. The goal is to style only the restricted mode. In the standard mode, the style should behave as a normal highlight.

> This means that you need some class on this span or on the entire editable that will allow you to target this style only for a single mode.

Fixed in https://github.com/ckeditor/ckeditor5-restricted-editing/pull/9 which has become main PR for the changes. I labeled this PR as a sub-pr.

**Comment by panr** [comment]
@Reinmar 🙏

**Pull request merged**